File: src/main/java/com/ejo/game/component/Background.java
package com.ejo.game.component;

import com.ejo.game.App;
import com.ejo.game.math.Vector;

import java.awt.*;
import java.util.ArrayList;
import java.util.Random;

public class Background implements Drawable {

    private final int starCount;
    private final int streakCount;

    private final ArrayList<Vector> starPosList = new ArrayList<>();
    private final ArrayList<Vector> streakPosList = new ArrayList<>();

    public Background(int starCount, int streakCount) {
        this.starCount = starCount;
        this.streakCount = streakCount;
        initStarPositions();
    }

    @Override
    public void draw(Graphics2D graphics, Vector pos) {
        Rectangle background = new Rectangle(App.WINDOW.getSize(), Color.BLACK);
        background.draw(graphics, Vector.NULL);

        for (int i = 0; i < starCount; i++) {
            Rectangle star = new Rectangle(new Vector(2,2), Color.WHITE,0);
            star.draw(graphics, starPosList.get(i));
        }

        for (int i = 0; i < streakCount; i++) {
            Rectangle star = new Rectangle(new Vector(20,2), Color.WHITE,Math.PI / 4);
            star.draw(graphics, streakPosList.get(i));
        }
    }

    private void initStarPositions() {
        Random random = new Random();
        for (int i = 0; i < starCount; i++) {
            starPosList.add(new Vector(random.nextInt((int)App.WINDOW.getSize().getX()), random.nextInt((int)App.WINDOW.getSize().getY())));
        }
        for (int i = 0; i < starCount; i++) {
            streakPosList.add(new Vector(random.nextInt((int)App.WINDOW.getSize().getX()), random.nextInt((int)App.WINDOW.getSize().getY())));
        }
    }

}

File: src/render/environment/Background.py
import math
import random

from pygame import Color

from src.render.base.Drawable import Drawable
from src.render.Rectangle import Rectangle, RotatedRectangle
from src.scene.Scene import Scene
from src.util.Vector import Vector

class Background(Drawable):

    def __init__(self, scene: Scene, window: 'Window', star_count: int, streak_count: int):
        super().__init__(scene, Vector.null())
        self.star_count = star_count
        self.streak_count = streak_count

        self.stars: list[Rectangle] = []
        self.streaks: list[Rectangle] = []

        #Generate Stars
        rand = random.Random()
        for i in range(star_count):
            pos: Vector = Vector(rand.randint(0,window.size.x),rand.randint(0,window.size.y))
            size: Vector = Vector(1,1)
            star: RotatedRectangle = RotatedRectangle(scene, pos, size, Color(255, 255, 255), 0)
            self.stars.append(star)

        #Generate Streaks
        for i in range(streak_count):
            pos: Vector = Vector(rand.randint(0,window.size.x),rand.randint(0,window.size.y))
            size: Vector = Vector(30,1)
            star: RotatedRectangle = RotatedRectangle(scene, pos, size, Color(255, 255, 255), math.tau / 8)
            self.streaks.append(star)
            self.streaks.append(star)

    def draw(self):
        #Draw Sky
        background: Rectangle = Rectangle(self.scene,Vector(0,0),self.scene.window.size,Color(0,0,0))
        background.draw()

        #Draw Stars
        for star in self.stars:
            star.draw()

        #Draw Streaks
        for streak in self.streaks:
            streak.draw()
